code/datums/diseases/parasitic_infection.dm

```diff
@@ -13,18 +13,28 @@
 	spread_flags = DISEASE_SPREAD_NON_CONTAGIOUS
 	required_organs = list(/obj/item/organ/liver)
 	bypasses_immunity = TRUE
+	var/obj/item/organ/liver/affected_liver
 
 
 /datum/disease/parasite/stage_act()
 	. = ..()
 	if(!.)
 		return
-
-	var/obj/item/organ/liver/affected_liver = affected_mob.getorgan(/obj/item/organ/liver)
 	if(!affected_liver)
+		affected_liver = affected_mob.getorganslot(ORGAN_SLOT_LIVER)
+	if(!affected_liver)
+		cure()
+		return FALSE
+	if(!affected_liver.owner)
```

Comment:
No it does something. Because it checks for a liver and links to it. The virus runs on the human but it registers the humans liver as a variable. When a organ doesnt have a owner then that implies that the liver is out in the open which is when the parasite dies.

Reply:
I learned a unsettling amount of stuff from my time turning naked nest into a removable organ.
Now that im looking at the before code i dont know if this virus was actually ever finished since it doesnt register the users liver at all. It just checks once if there is a liver.

Reply:
Maybe i should of just told you: Yes it works, yes i did test it, and looking at the rest of the code this was the only change i made and if it didnt work i would know it didnt work and would tell you.

Reply:
By before code i meant the code that existed before my current PR.